HW1/P5/P5_bfs.py

```diff
@@ -0,0 +1,18 @@
+# Takes in a src node and a destination node, and returns an rdd of
+# every shortest path from src to dest in the graph.
+# Caution: if src is not connected to dest, this runs forever
+def find_all_shortest_paths(src, dest, graph, sc):
+  accum = sc.accumulator(0)
+  num_parts = 64
+  graph = graph.partitionBy(num_parts).cache()
+  distances = sc.parallelize([(src, [src])]).partitionBy(num_parts)
+  while 1:
+    accum.value = 0
+    print "new iteration"
+    assert(distances.partitioner == graph.partitioner)
+    distances = distances.join(graph).map(lambda (k, v): (v[1], v[0] + [v[1]])).partitionBy(num_parts)
+    test = distances.filter(lambda (k, v): k == dest)
+    test.foreach(lambda x: accum.add(1))
```

Comment:
I think you could have used a map instead of filter, and increase the accumulator for the nodes with the distance == dest.